src/main/scripts/run_cdap_e2e_test.py

```diff
@@ -0,0 +1,134 @@
+# Copyright © 2023 Cask Data, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import os
+import subprocess
+import zipfile
+import shutil
+import argparse
+import re
+import configparser
+
+
+def run_shell_command(cmd):
+    process = subprocess.run(cmd.split(" "), stderr=subprocess.PIPE)
+    if process.returncode != 0:
+        print("Process completed with error: ", process.stderr)
+    assert process.returncode == 0
+
+
+def get_sandbox_version(directory_path):
+    pattern = re.compile(r"cdap-sandbox-(\d+\.\d+\.\d+)-SNAPSHOT.zip")
+    # Iterate over files and folders in the directory
+    for entry in os.listdir(directory_path):
+        if pattern.match(entry):
+            version_match = pattern.search(entry)
+            if version_match:
+                version = version_match.group(1)
+                return version
+
+
+def get_hydrator_branch(gitmodules_path, submodule_path):
+    config = configparser.ConfigParser()
+    config.read(gitmodules_path)
+
+    # Check if the submodule exists in the .gitmodules file
+    if config.has_section(f"submodule \"{submodule_path}\""):
+        branch = config.get(f"submodule \"{submodule_path}\"", "branch")
+        return branch
+    else:
+        return None
+
+
+# Parse command line optional arguments
+parser = argparse.ArgumentParser()
+parser.add_argument('--testRunner', help='TestRunner class to execute tests')
+parser.add_argument('--cdapBranch', help='CDAP branch against which PR is raised')
+args = parser.parse_args()
+
+print("Building CDAP Sandbox")
+os.chdir("./plugin")
+run_shell_command("git submodule update --init --recursive --remote")
+run_shell_command("mvn clean install -DskipTests -P templates,spark-dev")
+my_env = os.environ.copy()
+my_env["MAVEN_OPTS"] = "-Xmx1024m -XX:MaxPermSize=128m"
+my_env["_JAVA_OPTIONS"] = "-Xmx32G"
+
+# Building the plugins
+os.chdir("..")
+# Extracting hydrator-plugins branch from cdap-build gitmodules.
+cdap_branch = ""
+if args.cdapBranch:
```

Comment:
if `cdapBranch` is not present should we fail the workflow?

Reply:
Yes, that makes sense. Added a check to fail the workflow in case the cdapBranch is empty (fdbc4155ceb3c6e2bf31dc78a2352d2c09914397)